game/plugins/src/main/kotlin/gg/rsmod/plugins/content/items/jewellery/combat_bracelet.plugin.kts

```diff
@@ -0,0 +1,107 @@
+package gg.rsmod.plugins.content.items.jewellery
+
+import gg.rsmod.game.model.Tile
+import gg.rsmod.plugins.api.cfg.Items
+import gg.rsmod.plugins.content.magic.TeleportType
+import gg.rsmod.plugins.content.magic.canTeleport
+import gg.rsmod.plugins.content.magic.teleport
+
+val COMBAT_BRACELET = intArrayOf(
+    Items.COMBAT_BRACELET_1, Items.COMBAT_BRACELET_2, Items.COMBAT_BRACELET_3, Items.COMBAT_BRACELET_4
+)
+
+private val SOUNDAREA_ID = 200
+private val SOUNDAREA_RADIUS = 5
+private val SOUNDAREA_VOLUME = 1
+
+private val LOCATIONS = mapOf(//TODO CHANGE TO CORRECT COORDS
+    "Champions' Guild" to Tile(3086, 3503, 0),
+    "Warriors' Guild" to Tile(2917, 3175, 0),
+    "Monastery" to Tile(3104, 3249, 0),
+    "Ranging Guild" to Tile(3293, 3162, 0),
+)
+
+COMBAT_BRACELET.forEach { item ->
+    on_item_option(item = item, option = "Rub") {
+        player.queue {
+            when (options("Champions' Guild.", "Warriors' Guild.", "Monastery.", "Ranging Guild.", "Nowhere.")) {
+                1 -> player.teleport(LOCATIONS["Champions' Guild"]!!, isEquipped = false)
+                2 -> player.teleport(LOCATIONS["Warriors' Guild"]!!, isEquipped = false)
+                3 -> player.teleport(LOCATIONS["Monastery"]!!, isEquipped = false)
+                4 -> player.teleport(LOCATIONS["Ranging Guild"]!!, isEquipped = false)
+            }
+        }
+    }
+}
+
+COMBAT_BRACELET.forEach { item ->
+    LOCATIONS.forEach { (location, endTile) ->
+        on_equipment_option(item, option = location) {
+            player.queue(TaskPriority.STRONG) {
+                player.teleport(endTile, isEquipped = true)
+            }
+        }
+    }
+}
+
+/**
+ * Teleports the player using jewellery and updates the item's charges.
+ *
+ * @param endTile The destination tile for the player after teleportation.
+ * @param isEquipped Indicates if the item is equipped on the player (true) or in the inventory (false).
+ */
+fun Player.teleport(endTile: Tile, isEquipped: Boolean) {
+// Check if the player can teleport using items
+    if (canTeleport(TeleportType.JEWELRY)) {
+// Play a sound effect in the area around the player
+        world.spawn(AreaSound(tile, SOUNDAREA_ID, SOUNDAREA_RADIUS, SOUNDAREA_VOLUME))
+
+// Get the replacement bracelet with updated charges
+        val replacement = replacement(getInteractingItemId())
+
+        if (isEquipped) {
+// Update the equipped item, or remove it if there are no charges left
+            equipment[EquipmentType.GLOVES.id] = if (replacement > -1) Item(replacement) else null
+        } else {
+// Remove the item from the inventory and add the replacement if there are charges left
+            inventory.remove(getInteractingItem())
+            if (replacement > -1) {
+                inventory.add(replacement)
+            }
+        }
+
+// Teleport the player to the destination tile
```

Comment:
Please format code using the keybind `CTRL + ALT + L`, it's recommended to use `CTRL + A` to select all the code first.